chain/chunks/src/chunk_cache.rs

```diff
@@ -127,4 +137,73 @@ impl EncodedChunksCache {
             }
         }
     }
+
+    pub fn insert_chunk_header(&mut self, shard_id: ShardId, header: ShardChunkHeader) {
+        let height = header.inner.height_created;
+        if height
+            >= self.largest_seen_height.saturating_sub(NUM_BLOCK_HASH_TO_CHUNK_HEADER as BlockIndex)
```

Comment:
The number of heights matching this condition is `NUM_BLOCK_HASH_TO_CHUNK_HEADER + MAX_HEIGHTS_AHEAD + 1` but the size of the cache only fits `NUM_BLOCK_HASH_TO_CHUNK_HEADER`. So under normal circumstances one can easily make a particular node lose all the chunks by just sending chunks for 10 heights that do not include the next height.

I would also make the size of the cache couple times bigger (like 3x) because it doesn't hurt :) So that if someone does double sign at some point, and there are more than one header per height, the cache continue operating.